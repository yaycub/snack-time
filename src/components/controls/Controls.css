.Controls {
  text-align: center;
  button {
    padding: 0.4rem;
    margin: 0 0.4rem;

    border:none;

    background-color: #0198E1;
    color: #FFFFFF;
  }
}
