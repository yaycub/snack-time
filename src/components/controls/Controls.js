import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls = ({ actions, handleSelection }) => (
  <section className={styles.Controls}>
    {actions.map(({ name, text, count }) => (
      <button key={name} onClick={() => handleSelection(name)}>
        {text || name} {!!count && `- ${count}`}
      </button>
    ))}
  </section>
);

Controls.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    text: PropTypes.string
  })).isRequired,
  handleSelection: PropTypes.func.isRequired
};

export default Controls;
