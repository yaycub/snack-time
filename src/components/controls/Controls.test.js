import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls component', () => {
  let wrapper;
  let handleSelection;
  beforeEach(() => {
    const actions = [
      { name: 'DO_STUFF', text: 'stuff' },
      { name: 'other' }
    ];
    handleSelection = jest.fn();

    wrapper = shallow(<Controls
      actions={actions}
      handleSelection={handleSelection} />);
  });

  it('renders Controls', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handlesSelection with the action name', () => {
    wrapper.findWhere(el => el.key() === 'DO_STUFF').simulate('click');
    expect(handleSelection).toHaveBeenCalledWith('DO_STUFF');
  });
});
  
