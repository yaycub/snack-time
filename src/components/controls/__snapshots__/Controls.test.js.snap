// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<section
  className="Controls"
>
  <button
    key="DO_STUFF"
    onClick={[Function]}
  >
    stuff
  </button>
  <button
    key="other"
    onClick={[Function]}
  >
    other
  </button>
</section>
`;
